pallets/merkle/src/utils/hasher.rs

```diff
@@ -22,7 +42,12 @@ use sp_std::prelude::*;
 
 lazy_static! {
 	static ref DEFAULT_POSEIDON_HASHER: Poseidon = default_bulletproofs_poseidon_hasher();
-	// static ref POSEIDON_PARAMETERS: PoseidonParameters =
+	static ref POSEIDON_PARAMETERS: PoseidonParameters<Bls381> = setup_params_3::<Bls381>();
+	static ref PROVING_KEY: ProvingKey<Bls12_381> = {
```

Comment:
Proving key should not be defined anywhere on-chain IMO only the verifying key.

Reply:
Verifier key will be kept in storage for each backend. It might be saved in lazy_static in the future, but first I need more details about the setup ceremony and the results